bin/demo.py

```diff
@@ -2,221 +2,215 @@
-    command = "dcos package install --yes --options=tmp/jenkins.json jenkins"
-    print ("\n> " + command)
-    if call (['dcos', 'package', 'install', '--yes', '--options=tmp/jenkins.json', 'jenkins']) != 0:
-        log_and_exit ("Failed to install Jenkins.")
-    log("Jenkins has been installed! Wait for it to come up before proceeding at: {}".format(jenkins_url))
-
-def verify(jenkins_url):
-    r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-    if r.status_code != 200:
-        log ("Couldn't find a Jenkins instance running at {}.".format(jenkins_url))
+    dcos.config.set_val('core.dcos_url', dcos_url)
+    dcos.config.set_val('core.ssl_verify', 'False')
+
+def install_jenkins(jenkins_name, jenkins_url):
+    log("installing Jenkins with name '{}'".format(jenkins_name))
+    with open("conf/jenkins.json") as options_file:
+        package_config = options_file.read().replace("JENKINS_NAME", jenkins_name)
+    with open("jenkins_config.json", "w") as options_file:
+        options_file.write(package_config)
+    install_package('jenkins', None, jenkins_name, "jenkins_config.json")
+    os.remove("jenkins_config.json")
+    assert package_installed('jenkins', jenkins_name), log_and_exit('!! package failed to install')
+    log("waiting for Jenkins service to come up at '{}'".format(jenkins_url))
+    end_time = time.time() + 300
+    while time.time() < end_time:
+        if verify_jenkins(jenkins_url):
+            break
+        time.sleep(1)
+
+def verify_jenkins(jenkins_url):
+    try:
+        r = http.get(jenkins_url)
+        if r.status_code == 200 and r.headers['x-jenkins']:
+            log("service is up and running, got Jenkins version '{}'".format(r.headers['x-jenkins']))
+            return True
+    except:
         return False
-    log ("Jenkins is up and running! Got Jenkins version {}".format(r.headers['x-jenkins']))
-    return True
 
-def check_marathon_lb(elb_url):
-    log("Checking to see if Marathon-lb is installed.")
-    try:
-        r = requests.get(elb_url, verify=False)
-        if r.status_code == 503 and not r.text:
-            install_marathon_lb(elb_url)
-        else:
-            log ("Marathon-lb already seems to be running at {}. Not installing Marathon-lb.".format(elb_url))
-    except requests.exceptions.ConnectionError:
-        install_marathon_lb(elb_url)
-
-def install_marathon_lb(elb_url):
-    log ("Couldn't find a Marathon-lb instance running at {}.".format(elb_url))
-    log ("Installing Marathon-lb.")
-    command = "dcos package install --yes marathon-lb"
-    print ("\n> " + command)
-    if call (['dcos', 'package', 'install', '--yes', 'marathon-lb']) != 0:
-        log ("Failed to install Marathon-lb.")
-    else:
-        log("Marathon-lb has been installed!")
+def install_marathon_lb(marathon_lb_url):
+    log("installing marathon-lb")
+    with stdchannel_redirected(sys.stdout, os.devnull):
+        run_dcos_command('marathon app add conf/get_sa.json')
```

Comment:
Thought about this some more - we should maintain compatibility with Open DC/OS (since we also use this and the Tweeter demo to perform some smoke testing for DC/OS issues). Check if using Enterprise DC/OS and if not, set up the service account?